src/GraphQL.hs

```diff
@@ -135,6 +142,75 @@ interpretAnonymousQuery
   -> m Response -- ^ The result of running the query.
 interpretAnonymousQuery handler query = interpretQuery @api @m handler query Nothing mempty
 
+data SchemaRoot m query mutation = SchemaRoot
+  { queries   :: Handler m query
+  , mutations :: Handler m mutation
+  }
+
+-- | Execute a query or mutation
+--
+-- Similar to executeQuery, execept requests are dispatched against the
+-- SchemaRoot depending on whether they are a query or mutation
+executeRequest
+  :: forall schema queries mutations m.
+  ( schema ~ SchemaRoot m queries mutations
+  , HasResolver m queries
+  , HasObjectDefinition queries
+  , HasResolver m mutations
+  , HasObjectDefinition mutations
+  , Monad m
+  )
+  => SchemaRoot m queries mutations
+  -> QueryDocument VariableValue
+  -> Maybe Name
+  -> VariableValues
+  -> m Response
+executeRequest (SchemaRoot qh mh) document name variables =
+  case getOperation document name variables of
+    Left e -> pure (ExecutionFailure (singleError e))
+    Right (operation, ss) -> do
+      toResult <$> case operation of
+        Query    _ _ _ -> resolve @m @queries qh (Just ss)
+        Mutation _ _ _ -> resolve @m @mutations mh (Just ss)
+
+-- | Interpret a query or mutation against a SchemaRoot
+interpretRequest
```

Comment:
What's the difference between `interpretRequest` and `executeRequest`? Is `interpretRequest` including the parsing of the document?